web_ui/frontend/app/registry/namespace/register/page.tsx

```diff
@@ -0,0 +1,113 @@
+ *
+ * Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+"use client"
+
+import {
+    Box,
+    Button,
+    Grid,
+    Typography,
+    TextField,
+    FormControl,
+    InputLabel,
+    Select,
+    MenuItem,
+    TextareaAutosize,
+    FormHelperText,
+    Collapse,
+    Alert
+} from "@mui/material";
+import React, {ReactNode, useEffect, useMemo, useState} from "react";
+
+import Link from "next/link";
+
+import {Alert as AlertType} from "@/components/Main";
+import NamespaceForm from "@/app/registry/namespace/components/NamespaceForm";
+import AuthenticatedContent from "@/components/layout/AuthenticatedContent";
+import {secureFetch} from "@/helpers/login";
+
+export default function Register() {
+
+    const [alert, setAlert] = useState<AlertType | undefined>(undefined)
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) : Promise<boolean> => {
+
+        e.preventDefault()
+
+        const formData = new FormData(e.currentTarget);
+
+        try {
+            const response = await secureFetch("/api/v1.0/registry_ui/namespaces", {
+                body: JSON.stringify({
+                    prefix: formData.get("prefix"),
+                    pubkey: formData.get("pubkey"),
+                    admin_metadata: {
+                        description: formData.get("description"),
+                        site_name: formData.get("site-name"),
+                        institution: formData.get("institution"),
+                        security_contact_user_id: formData.get("security-contact-user-id")
+                    }
+                }),
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                credentials: "include"
+            })
+
+            if(!response.ok){
+                try {
+                    let data = await response.json()
+                    setAlert({severity: "error", message: response.status + ": " + data['error']})
+                } catch (e) {
+                    setAlert({severity: "error", message: `Failed to register namespace: ${formData.get("prefix")}`})
+                }
+            } else {
+                setAlert({severity: "success", message: `Successfully registered namespace: ${formData.get("prefix")}`})
```

Comment:
Can we redirect user to `/registry` page here after the alert? It's a bit weird to stay on an empty registration form once the user successfully submit their registration.

Reply:
@haoming29 That was intentional incase they are looking to submit multiple entries at once. If you don't see that happening I will redirect.

Reply:
My guess is no, unless people who manage a bunch of namespaces from topology decide to switch many of them over to Pelican.

Reply:
> My guess is no, unless people who manage a bunch of namespaces from topology decide to switch many of them over to Pelican.

In that case, I would recommend them to write their little python script to do it. It's going to be tedious and error prone to do it manually.